src/app/services/baw-api/base-api.service.ts

```diff
@@ -62,126 +64,72 @@ export abstract class BawApiService {
-    if (!filters) {
-      this.httpGet<APIResponse>(path).subscribe(
-        this.handleResponse(next, error)
-      );
-    } else {
-      this.httpPost<APIResponse>(path + "/filter", filters).subscribe(
-        this.handleResponse(next, error)
-      );
-    }
-  }
-
-  /**
-   * Create request for API route
-   * @param next Callback function for successful response
-   * @param error Callback function for failed response
-   * @param path API path
-   * @param body Request body
-   */
-  protected apiCreate(
-    next: (data: any) => void,
-    error: (err: any) => void,
-    path: string,
-    body?: any
-  ) {
-    this.httpPost<APIResponse>(path, body).subscribe(
-      this.handleResponse(next, error)
-    );
-  }
-
-  /**
-   * Update request for API route
-   * @param next Callback function for successful response
-   * @param error Callback function for failed response
-   * @param path API path
-   * @param body Request body
-   */
-  protected apiUpdate(
+  protected apiRequest(
+    method: "LIST" | "FILTER" | "SHOW" | "CREATE" | "UPDATE" | "DELETE",
     next: (data: any) => void,
     error: (err: any) => void,
     path: string,
     body?: object
   ) {
-    this.httpPatch<APIResponse>(path, body).subscribe(
-      this.handleResponse(next, error)
-    );
+    path = this.url + path;
+
+    switch (method) {
+      case "LIST":
+      case "SHOW":
+        return this.http
+          .get<APIResponse>(path)
+          .subscribe(this.handleResponse(next, error));
+      case "FILTER":
+        return this.http
+          .post<APIResponse>(path + "/filter", body)
+          .subscribe(this.handleResponse(next, error));
+      case "CREATE":
+        return this.http
+          .post<APIResponse>(path, body)
+          .subscribe(this.handleResponse(next, error));
+      case "UPDATE":
+        return this.http
+          .patch<APIResponse>(path, body)
+          .subscribe(this.handleResponse(next, error));
+      case "DELETE":
+        return this.http
+          .delete<APIResponse>(path)
+          .subscribe(this.handleResponse(next, error, true));
+    }
   }
 
   /**
-   *
-   * @param next Callback function for successful response
-   * @param error Callback function for failed response
-   * @param path API path
+   * Templates a string by substituting placeholders for tokens later in execution.
+   * It is designed to work as the tag function for tagged interpolated strings.
+   * @returns A reusable template function that is statically checked for arity and
+   * parameter type compatibility with the placeholders from the interpolated string.
+   * @param strings The strings around the tokens to template
+   * @param placeholders Placeholders that are substituted for token when
+   *  templating is done. Note these are transform functions.
    */
-  protected apiDelete(
-    next: (data: any) => void,
-    error: (err: any) => void,
-    path: string
+  protected makeTemplate<T extends ((any: any) => any)[]>(
```

Comment:
makeTemplate is general enough to use anywhere. Put it in its own file, perhaps `stringTemplating.ts`